Automation hub, team and org scoped automations (#2959)
wandb/docs: models/automations.mdx

import AutomationsBestPractices from "/snippets/_includes/automations/best-practices.mdx";
import AutomationsWhereToFind from "/snippets/_includes/automations/where-to-find-automations.mdx";

> You can scope an automation to an **Organization**, a **Team**, a **Registry**, a **Registry collection**, or a **project**. An automation observes the event across everything in its scope, including projects, registries, and collections created later. Where you create an automation, which events you can use, and how scope works all differ. For event types by scope, see [Automation events and scopes](/models/automations/automation-events).

Author: ```suggestion
_Automations_ send Slack notifications or run webhooks in response to specific events in W&B. You can scope an automation to an organization, a team, project, registry, or collection. An automation observes the event across everything within its scope, including projects, registries, and collections created later. The steps to create an automation and the available events depend on the scope. For event types by scope, see [Automation events and scopes](/models/automations/automation-events).
```

<AutomationsWhereToFind/>

## Automations hub

The **Automations** hub is a global surface in the left navigation that aggregates every automation you can access, across all scopes and projects, in a single list. Use it to discover, inspect, and manage automations without navigating to each project or registry.

From the **Automations** hub, you can:

- View all automations you can access in one list, then filter and sort them (for example, by scope, trigger, or action).

> - Use the quick filter to switch between automations you created and all automations in your organization.
> - Click an automation's name to view its full configuration in a details drawer.
> - View an automation's [execution history](/models/automations/view-automation-history) to check whether it succeeded or failed.
> - Create an automation and choose its scope, without needing to know the scoped entry point first.
> - Select multiple automations and delete them at once.
> 

Author: ```suggestion
- Create an automation and choose its scope, without needing to know the scoped entry point first.
- View all automations you can access in one list, then filter and sort them (for example, by scope, trigger, or action).
- Click an automation's name to view its full configuration details.
- View an automation's [execution history](/models/automations/view-automation-history) to check whether it succeeded or failed.
- Use the quick filter to switch between automations you created and all automations in your organization.
- Select multiple automations and delete them at once.

```

> The **Automations** hub complements the **Automations** tab in each project and registry; those scoped tabs remain available for quick access.

Author: ```suggestion
The **Automations** hub complements the **Automations** tab in each project and registry; those scoped tabs remain available for quick targeted access.
```

<AutomationsBestPractices/>

## Limitations

> - [Run metric automations](/models/automations/automation-events/#run-metrics-change) and [run metrics z-score change automations](/models/automations/automation-events/#run-metrics-z-score-change) are supported only in [W&B Multi-tenant Cloud](/platform/hosting/#wb-multi-tenant-cloud).

Author: ```suggestion
- [Run metric automations](/models/automations/automation-events/#run-metrics-change) and [run metrics z-score change automations](/models/automations/automation-events/#run-metrics-z-score-change) are supported only in [W&B Multi-tenant Cloud](/platform/hosting/#wb-multi-tenant-cloud) and [Dedicated Cloud](/platform/hosting/dedicated-cloud).
```

> - [Run metric automations](/models/automations/automation-events/#run-metrics-change) and [run metrics z-score change automations](/models/automations/automation-events/#run-metrics-z-score-change) are supported only in [W&B Multi-tenant Cloud](/platform/hosting/#wb-multi-tenant-cloud) and [Dedicated Cloud](/platform/hosting/dedicated-cloud).

Author: ```suggestion
- [Run metric automations](/models/automations/automation-events/#run-metrics-change) and [run metrics z-score change automations](/models/automations/automation-events/#run-metrics-z-score-change) are supported only in [W&B Multi-tenant Cloud](/platform/hosting/#wb-multi-tenant-cloud) and [Dedicated Cloud](/platform/hosting/hosting-options/dedicated-cloud).
```

> - At the **Team** and **Organization** scopes, the W&B App supports only artifact and collection events. Run-based events aren't available in the App at these scopes. See [Automation events and scopes](/models/automations/automation-events).

Reviewer: Not sure if you mention this elsewhere, but it **_is_** possible to create team-/org-scoped run automations programmatically (via the Automations API in the SDK).  At least it will be -- pending the following PR (still in review): https://github.com/wandb/wandb/pull/12194

**suggestion:** May be worth mentioning the programmatic (SDK) API **does** allow these, to avoid undue confusion

Extra context:
- Run automations aren't outright disabled for team/org scopes, they're just hidden in the (more curated) UI
- The fact that we hid them in the UI was a late-game decision. Not essential for this PR, but for reference, some context for the decision: [thread](https://coreweave.enterprise.slack.com/archives/C0AVC9YKZH7/p1783458196899399?thread_ts=1783458027.467879&channel=C0AVC9YKZH7&message_ts=1783458196.899399)

Author: Discussed in Slack, we will add these details once the fix is in a released SDK version. :)

<AutomationsBestPractices/>

## Limitations

> - [Run metric automations](/models/automations/automation-events/#run-metrics-events) and [run metrics z-score change automations](/models/automations/automation-events/#run-metrics-z-score-change-automations) are supported only in [W&B Multi-tenant Cloud](/platform/hosting/#wb-multi-tenant-cloud).

Author: Good catch — these were pre-existing broken anchors that I inherited when I reworked this line. Fixed in a7f01cb: `#run-metrics-events` → `#run-metrics-change` and `#run-metrics-z-score-change-automations` → `#run-metrics-z-score-change` (both now resolve; `mint broken-links` passes clean). Also fixed the "webhoook" typo you flagged in Next steps.
